packages/dds/tree/src/feature-libraries/chunked-forest/uniformChunk.ts

```diff
@@ -519,7 +532,14 @@ class Cursor extends SynchronousCursor implements ChunkedCursor {
 	}
 
 	public get value(): Value {
+		const idCompressor = this.chunk.idCompressor;
 		const info = this.nodeInfo(CursorLocationType.Nodes);
+		if (info.shape.hasValue && info.shape.maybeDecompressedStringAsNumber) {
+			const value = this.chunk.values[info.valueOffset];
+			if (typeof value === "number" && idCompressor !== undefined) {
```

Comment:
does this if statement guarantee that the value was a compressed string? if not, we should use tryDecompress below.

Reply:
I wasn't able to find the `tryDecompress` method (although `tryRecompress` exists), so I wasn't sure what the best way to handle this was.

However, the idea was that we will only set the `maybeDecompressedStringAsNumber` flag on string leaf nodes, so that it would be impossible for the value to be a number at this point unless it was already compressed. This would be done in the follow-up PR where we make sure that we check the schema and set this flag for all string leaf node shapes before passing it down to `uniformChunkFromCursor` (or any other place we need to create a UniformChunk with encoded ids)

Would it be best to just wrap this in a try catch block and throw an error to catch this case if it ever happens?

Reply:
Ah, right, I forgot we decided not to add that flavor because if you're doing a cast that you aren't sure is correct, it's code smell. However, it sounds like you are certain in this case, but the code doesn't really make that obvious (do the docs?). Why is the variable named "maybe"? Should we be asserting that the value is always a number instead of conditionally decompressing if it is? I guess the contract isn't really clear here.

Reply:
The flag was named with "maybe" since some string leaf nodes that may contain strings that are either not stable ids, or unknown stable ids. But I realized that there's nothing stopping anyone from setting this flag when creating shapes for non string leaf nodes currently. I have now added a check to throw a usage error if this happens, and updated the docs :)

I also added another test to test for this as well.